src/layouts/sidebar-sidecar/server.ts

```diff
@@ -118,6 +120,13 @@ export function getStaticGenerationFunctions<
       const pathParts = (ctx.params.page || []) as string[]
       const headings = [] // populated by anchorLinks plugin below
 
+      // catch invalid URIs early
+      if (isInvalidURI(pathParts.join('/'))) {
+        return {
+          notFound: true,
+        }
+      }
```

Comment:
💭 Can we expand on the approach / heuristic here for how we determine an invalid URL / why we are doing this? I think that will be a helpful breadcrumb for the future!

Reply:
Actually thinking about this more, i'm curious the reasoning behind catching **only** `404`'s and returning `notFound: true`...

https://github.com/hashicorp/dev-portal/blob/f8cf9c78dfd42045d09d0744b9606693c7de0c6e/src/layouts/sidebar-sidecar/server.ts#L163-L166

This class of bad requests cause our Content API to return a `Vercel 400`, so maybe it'd be simpler to include that in the detected status codes